python/ray/train/examples/torch_data_prefetch_benchmark/README.rst

```diff
@@ -0,0 +1,32 @@
+Torch Data Prefetching Benchmark Example
+========================================
+
+We provide a benchmark example to show how the auto pipeline for host to device data transfer speeds up training on GPUs.
+Running the following command gives the actual runtime of a small model training with and without the auto pipeline functionality.
+The experiment size can be modified by setting different values for :code:`epochs` and :code:`num_hidden_layers`, e.g.,
+
+.. code-block:: bash
+
+    python auto_pipeline_for_host_to_device_data_transfer.py --epochs 2 --num_hidden_layers 2
+
+
+The table below displays the runtime (excluding preparation work) under different configurations.
+The first value in the parentheses reports the runtime of using the auto pipeline, and the second reports the time of not using it.
+These experiments were done on a NVIDIA 2080 Ti.
+The auto pipeline functionality offers more speed improvement when the model size and the number of epochs get larger.
+(The actual runtime outputs may vary if these experiments are run locally or different hardware devices are used.)
+
+|
++-------------+---------------------+---------------------+----------------------+
+|             |                      number of hidden layers                     |
++-------------+---------------------+---------------------+----------------------+
+|             |           1         |           4         |           8          |
++=============+==========+==========+==========+==========++==========+==========+
+| epoch = 1   | 2.52s    | 2.69s    | 6.85s    | 7.21s    | 13.05s    | 13.54s   |
++-------------+----------+----------+----------+----------+-----------+----------+
+| epoch = 5   | 12.14s   | 12.88s   | 34.33s   | 36.48s   | 66.38s    | 69.12s   |
++-------------+----------+----------+----------+----------+-----------+----------+
+| epoch = 50  | 123.12s  | 132.88s  | 369.42s  | 381.67s  | 693.52s   | 736.17s  |
++-------------+----------+----------+----------+----------+-----------+----------+
```

Comment:
Hmm, yeah it doesn't look like tables render very well in our docs. I wonder if it'll be cleaner to reduce the number of columns. 

Example:
```
                             | False |  True
================================================
(epochs = 1, num_layers = 1) |  ...
(epochs = 1, num_layers = 4) |  ...
(epochs = 1, num_layers = 8) |  ...
(epochs = 5, num_layers = 1) |  ...
(epochs = 5, num_layers = 4) |  ...
(epochs = 5, num_layers = 8) |  ...
(epochs = 50, num_layers = 1) |  ...
(epochs = 50, num_layers = 4) |  ...
(epochs = 50, num_layers = 8) |  ...
```